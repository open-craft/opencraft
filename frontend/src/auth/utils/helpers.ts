import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
