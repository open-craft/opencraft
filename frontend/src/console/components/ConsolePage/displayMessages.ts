const messages = {
  editCourses: {
    defaultMessage: 'Edit courses (Studio)',
    description: 'Label to Studio link.'
  },
  notAllowedForStaff: {
    defaultMessage:
      'This page is not accessible by staff users yet. Use <link>this link</link> instead.',
    description: 'Error message for staff users'
  }
};

export default messages;
