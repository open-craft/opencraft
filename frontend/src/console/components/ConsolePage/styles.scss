@import "~styles/theme";

.console-page {
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: stretch;

  .title-container {
    margin: 75px;

    @media only screen and (max-width: 600px)
    {
      margin: 30px;
    }
  }

  h1, h2 {
    text-align: center;
    color: #fff;
  }

  h1 {
    font-family: $playfair-font;
    font-weight: bold;
    font-size: 40px;
  }

  h2 {
    font-family: $chivo-font;
    font-size: 20px;
  }

  .console-page-container {
    background-color: #fff;
    padding: 3rem 1rem;
    margin: 0 -15px;
    flex: 1;

    .console-page-content {
      margin: 0 auto;
      max-width: 1100px;
      justify-content: center;

      .customization-form {
        background-color: #f0f5f7;
        padding: 32px;
        margin-bottom: 16px;

        .loading {
          text-align: center;
          margin-top: 80px;
          margin-bottom: auto;
        }

        h2 {
          font-family: $chivo-font;
          font-size: 24px;
          letter-spacing: 0.72px;
          color: $secondary-1;
          text-align: left;
          margin-bottom: 32px;
        }
      }
    }
  }
}



.header-link {
  color: white;

  &:hover {
    color: white;
  }

  .instance-link {
    margin-left: 15px;
  }
}
