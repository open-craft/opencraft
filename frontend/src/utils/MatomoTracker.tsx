import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
