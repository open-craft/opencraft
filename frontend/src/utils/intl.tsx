import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
