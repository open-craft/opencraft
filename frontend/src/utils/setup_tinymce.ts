import 'tinymce/tinymce';
import 'tinymce/themes/silver';
import 'tinymce/skins/ui/oxide/skin.min.css';
import 'tinymce/skins/ui/oxide/content.min.css';
import 'tinymce/plugins/advlist';
import 'tinymce/plugins/autolink';
import 'tinymce/plugins/lists';
import 'tinymce/plugins/link';
import 'tinymce/plugins/image';
import 'tinymce/plugins/charmap';
import 'tinymce/plugins/print';
import 'tinymce/plugins/preview';
import 'tinymce/plugins/anchor';
import 'tinymce/plugins/searchreplace';
import 'tinymce/plugins/visualblocks';
import 'tinymce/plugins/code';
import 'tinymce/plugins/fullscreen';
import 'tinymce/plugins/insertdatetime';
import 'tinymce/plugins/media';
import 'tinymce/plugins/table';
import 'tinymce/plugins/paste';
import 'tinymce/plugins/help';
import 'tinymce/plugins/wordcount';
