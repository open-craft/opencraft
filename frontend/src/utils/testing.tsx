import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
