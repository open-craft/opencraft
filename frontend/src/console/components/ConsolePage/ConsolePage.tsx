import * as React from 'react';
import { INTERNAL_DOMAIN_NAME } from 'global/constants';
import { RedeploymentToolbar, CustomizationSideMenu } from 'console/components';
import { Row, Col, Container } from 'react-bootstrap';
import { WrappedMessage } from 'utils/intl';
import { InstancesModel } from 'console/models';
import { RootState } from 'global/state';
import { connect } from 'react-redux';
import {
  listUserInstances,
  getDeploymentStatus,
  performDeployment,
  cancelDeployment
} from 'console/actions';
import messages from './displayMessages';

import './styles.scss';

interface ActionProps {
  cancelDeployment: Function;
  getDeploymentStatus: Function;
  listUserInstances: Function;
  performDeployment: Function;
}

interface StateProps extends InstancesModel {}
interface Props extends StateProps, ActionProps {
  children: React.ReactNode;
  contentLoading: boolean;
}

export class ConsolePageComponent extends React.PureComponent<Props> {
  public componentDidMount() {
    if (!this.props.loading && this.props.activeInstance.data === null) {
      this.props.listUserInstances();
    }
    setInterval(() => {
      if (this.props.activeInstance.data) {
        this.props.getDeploymentStatus(this.props.activeInstance.data.id);
      }
    }, 5000);
  }

  private performDeployment() {
    if (this.props.activeInstance.data) {
      this.props.performDeployment(this.props.activeInstance.data.id);
    }
  }

  private cancelDeployment() {
    if (this.props.activeInstance.data) {
      this.props.cancelDeployment(this.props.activeInstance.data.id);
    }
  }

  private renderHeader() {
    if (this.props.loading || this.props.activeInstance.data === null) {
      return (
        <div className="title-container">
          <h1>
            <i className="fas fa-sync-alt fa-spin" />
          </h1>
        </div>
      );
    }

    const instanceData = this.props.activeInstance.data;
    const instanceLink =
      `https://${instanceData.subdomain}${INTERNAL_DOMAIN_NAME}` || '';
    const studioLink =
      `https://studio.${instanceData.subdomain}${INTERNAL_DOMAIN_NAME}` || '';

    return (
      <div className="title-container">
        <h1>
          <a className="header-link" href={instanceLink}>
            {instanceData.instanceName}
            <i className="instance-link fas fa-link fa-xs" />
          </a>
        </h1>
        <h2>
          <a className="header-link" href={studioLink}>
            <WrappedMessage messages={messages} id="editCourses" />
          </a>
        </h2>
      </div>
    );
  }

  public render() {
    const content = () => {
      if (this.props.contentLoading) {
        return (
          <div className="loading">
            <i className="fas fa-2x fa-sync-alt fa-spin" />
          </div>
        );
      }
      return this.props.children;
    };

    let deploymentLoading = true;
    if (this.props.activeInstance && this.props.activeInstance.loading) {
      deploymentLoading = this.props.activeInstance.loading.includes(
        'deployment'
      );
    }

    return (
      <div className="console-page">
        {this.renderHeader()}

        <RedeploymentToolbar
          deployment={
            this.props.activeInstance
              ? this.props.activeInstance.deployment
              : undefined
          }
          cancelRedeployment={() => {
            this.cancelDeployment();
          }}
          performDeployment={() => {
            this.performDeployment();
          }}
          loading={deploymentLoading}
        />

        <div className="console-page-container">
          <Row className="console-page-content">
            <Container fluid>
              <Row>
                <Col md="4">
                  <CustomizationSideMenu />
                </Col>
                <Col md="8" className="customization-form">
                  {content()}
                </Col>
              </Row>
            </Container>
          </Row>
        </div>
      </div>
    );
  }
}

export const ConsolePage = connect<
  StateProps,
  ActionProps,
  {},
  Props,
  RootState
>((state: RootState) => state.console, {
  cancelDeployment,
  getDeploymentStatus,
  listUserInstances,
  performDeployment
})(ConsolePageComponent);
