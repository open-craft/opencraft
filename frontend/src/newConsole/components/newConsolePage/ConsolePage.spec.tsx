import React from 'react';
import { setupComponentForTesting } from "utils/testing";
import { ConsolePage } from './ConsolePage';


describe("Console Page", function() {
  it('Correctly renders loading page', () => {
      const tree = setupComponentForTesting(
        <ConsolePage contentLoading={true}>
          <span> Test! </span>
        </ConsolePage>,
        {
          console: {
            loading: true
          }
        }
      ).toJSON();
      expect(tree).toMatchSnapshot();
  });

  it('Correctly renders page with data', () => {
      const tree = setupComponentForTesting(
        <ConsolePage contentLoading={false}>
          <span> Test! </span>
        </ConsolePage>,
        {
          console: {
            loading: false,
            activeInstance: {
              data: {
                id: 1,
                instanceName: "test",
                subdomain: "test",
                lmsUrl: "test-url",
                studioUrl: "test-url",
                isEmailVerified: true,
              },
              deployment: {
                status: "preparing",
                undeployedChanges: [],
                deployedChanges: null,
                type: 'admin',
              }
            },
            instances: [{
              id: 1,
              instanceName: "test",
              subdomain: "test",
            }]
          }
        }
      ).toJSON();
      expect(tree).toMatchSnapshot();
  });

  it('Correctly renders page with email not verified alert', () => {
    const tree = setupComponentForTesting(
      <ConsolePage contentLoading={false}>
        <span> Test! </span>
      </ConsolePage>,
      {
        console: {
          loading: false,
          activeInstance: {
            data: {
              id: 1,
              instanceName: "test",
              subdomain: "test",
              lmsUrl: "test-url",
              studioUrl: "test-url",
              isEmailVerified: false,
            },
            deployment: {
              status: "preparing",
              undeployedChanges: [],
              deployedChanges: null,
              type: 'admin',
            }
          },
          instances: [{
            id: 1,
            instanceName: "test",
            subdomain: "test",
          }]
        }
      }
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
});
