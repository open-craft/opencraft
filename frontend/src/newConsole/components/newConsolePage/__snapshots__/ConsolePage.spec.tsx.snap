// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Console Page Console Page with Edit component Correctly renders loading page 1`] = `
<div
  className="console-page"
>
  <div
    className="d-flex justify-content-between redeployment-toolbar px-4 py-3 container-fluid"
  >
    <div
      className="d-flex flex-row justify-content-between w-100"
    >
      <span
        className="status-pill badge badge-pill badge-primary"
        style={
          Object {
            "backgroundColor": "grey",
          }
        }
      >
        <div
          className="text"
        >
          Loading
        </div>
        <i
          className="fas fa-question-circle fa-lg"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        />
      </span>
      <button
        className="float-right btn-disabled btn btn-primary btn-lg"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        <div
          className="button-text-layer"
        >
          Publish changes
        </div>
      </button>
    </div>
    <div
      className="toast-container"
    />
  </div>
  <div
    className="new-console-page-container"
  >
    <div
      className="new-console-page-content row"
    >
      <div
        className="pr-4 pl-0 container-fluid"
      >
        <div
          className="justify-content-center m-0 row"
        >
          <div
            className="p-0 m-0 pl-4 col-md-3"
          >
            <button
              className="back-button btn btn-link btn-sm"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span>
                <i
                  aria-hidden="true"
                  className="fa fa-angle-left sm"
                />
              </span>
              <span>
                Back
              </span>
            </button>
            <div
              className="customization-form"
            >
              <div
                className="loading"
              >
                <i
                  className="fas fa-2x fa-sync-alt fa-spin"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Console Page Correctly renders loading page 1`] = `
<div
  className="console-page"
>
  <div
    className="d-flex justify-content-between redeployment-toolbar px-4 py-3 container-fluid"
  >
    <div
      className="d-flex flex-row justify-content-between w-100"
    >
      <span
        className="status-pill badge badge-pill badge-primary"
        style={
          Object {
            "backgroundColor": "grey",
          }
        }
      >
        <div
          className="text"
        >
          Loading
        </div>
        <i
          className="fas fa-question-circle fa-lg"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        />
      </span>
      <button
        className="float-right btn-disabled btn btn-primary btn-lg"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        <div
          className="button-text-layer"
        >
          Publish changes
        </div>
      </button>
    </div>
    <div
      className="toast-container"
    />
  </div>
  <div
    className="new-console-page-container"
  >
    <div
      className="new-console-page-content row"
    >
      <div
        className="pr-4 pl-0 container-fluid"
      >
        <div
          className="justify-content-center m-0 row"
        >
          <div
            className="p-0 m-0 col-md-3"
          >
            <div
              className="new-customization-menu accordion"
            >
              <div
                className="card"
              >
                <div
                  className="card-header"
                  onClick={[Function]}
                  variant="link"
                >
                  Theme
                </div>
                <div
                  aria-expanded={null}
                  className="collapse show"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/theming/preview-and-colors"
                        onClick={[Function]}
                      >
                        Colors
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/logos"
                        onClick={[Function]}
                      >
                        Logos
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/buttons"
                        onClick={[Function]}
                      >
                        Buttons
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/navigation"
                        onClick={[Function]}
                      >
                        Navigation
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/footer"
                        onClick={[Function]}
                      >
                        Footer
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/hero"
                        onClick={[Function]}
                      >
                        Hero
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="card"
              >
                <div
                  className="card-header"
                  onClick={[Function]}
                  variant="link"
                >
                  Custom Pages
                </div>
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/custom-pages/about"
                        onClick={[Function]}
                      >
                        About
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/contact"
                        onClick={[Function]}
                      >
                        Contact
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/donate"
                        onClick={[Function]}
                      >
                        Donate
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/tos"
                        onClick={[Function]}
                      >
                        Terms of Service
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/honor"
                        onClick={[Function]}
                      >
                        Honor Code
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/privacy"
                        onClick={[Function]}
                      >
                        Privacy Policy
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="card"
              >
                <div
                  className="card-header"
                  onClick={[Function]}
                  variant="link"
                >
                  Instance Settings
                </div>
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/settings/general"
                        onClick={[Function]}
                      >
                        General
                      </a>
                      <a
                        aria-current="page"
                        className="disabled active"
                        href="/"
                        onClick={[Function]}
                        style={Object {}}
                      >
                        Domain
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="card"
              >
                <div
                  className="card-header"
                  onClick={[Function]}
                  variant="link"
                >
                  Courses (Studio)
                </div>
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/courses/manage"
                        onClick={[Function]}
                      >
                        Manage
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="pr-0 col-md-9"
          >
            <div
              className="customization-form"
            >
              <div
                className="loading"
              >
                <i
                  className="fas fa-2x fa-sync-alt fa-spin"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Console Page Correctly renders page with data 1`] = `
<div
  className="console-page"
>
  <div
    className="d-flex justify-content-between redeployment-toolbar px-4 py-3 container-fluid"
  >
    <div
      className="d-flex flex-row justify-content-between w-100"
    >
      <span
        className="status-pill badge badge-pill badge-primary"
        style={
          Object {
            "backgroundColor": "#ff9b04",
          }
        }
      >
        <div
          className="text"
        >
          Publishing
        </div>
        <i
          className="fas fa-question-circle fa-lg"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        />
      </span>
      <button
        className="float-right btn-disabled btn btn-primary btn-lg"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        <div
          className="button-text-layer"
        >
          Publish changes
        </div>
      </button>
    </div>
    <div
      className="toast-container"
    />
  </div>
  <div
    className="new-console-page-container"
  >
    <div
      className="new-console-page-content row"
    >
      <div
        className="pr-4 pl-0 container-fluid"
      >
        <div
          className="justify-content-center m-0 row"
        >
          <div
            className="p-0 m-0 col-md-3"
          >
            <div
              className="new-customization-menu accordion"
            >
              <div
                className="card"
              >
                <div
                  className="card-header"
                  onClick={[Function]}
                  variant="link"
                >
                  Theme
                </div>
                <div
                  aria-expanded={null}
                  className="collapse show"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/theming/preview-and-colors"
                        onClick={[Function]}
                      >
                        Colors
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/logos"
                        onClick={[Function]}
                      >
                        Logos
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/buttons"
                        onClick={[Function]}
                      >
                        Buttons
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/navigation"
                        onClick={[Function]}
                      >
                        Navigation
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/footer"
                        onClick={[Function]}
                      >
                        Footer
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/hero"
                        onClick={[Function]}
                      >
                        Hero
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="card"
              >
                <div
                  className="card-header"
                  onClick={[Function]}
                  variant="link"
                >
                  Custom Pages
                </div>
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/custom-pages/about"
                        onClick={[Function]}
                      >
                        About
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/contact"
                        onClick={[Function]}
                      >
                        Contact
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/donate"
                        onClick={[Function]}
                      >
                        Donate
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/tos"
                        onClick={[Function]}
                      >
                        Terms of Service
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/honor"
                        onClick={[Function]}
                      >
                        Honor Code
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/privacy"
                        onClick={[Function]}
                      >
                        Privacy Policy
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="card"
              >
                <div
                  className="card-header"
                  onClick={[Function]}
                  variant="link"
                >
                  Instance Settings
                </div>
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/settings/general"
                        onClick={[Function]}
                      >
                        General
                      </a>
                      <a
                        aria-current="page"
                        className="disabled active"
                        href="/"
                        onClick={[Function]}
                        style={Object {}}
                      >
                        Domain
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="card"
              >
                <div
                  className="card-header"
                  onClick={[Function]}
                  variant="link"
                >
                  Courses (Studio)
                </div>
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/courses/manage"
                        onClick={[Function]}
                      >
                        Manage
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="pr-0 col-md-9"
          >
            <span>
               Test! 
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Console Page Correctly renders page with email not verified alert 1`] = `
<div
  className="console-page"
>
  <div
    className="d-flex justify-content-center align-middle email-activation-alert"
  >
    <i
      className="alert-icon fas fa-exclamation-triangle fa-xs"
    />
    One of your email addresses is pending confirmation. Please confirm it to begin managing your Open edX instance.
  </div>
  <div
    className="new-console-page-container"
  >
    <div
      className="new-console-page-content row"
    >
      <div
        className="pr-4 pl-0 container-fluid"
      >
        <div
          className="justify-content-center m-0 row"
        >
          <div
            className="p-0 m-0 col-md-3"
          >
            <div
              className="new-customization-menu accordion"
            >
              <div
                className="card"
              >
                <div
                  className="card-header"
                  onClick={[Function]}
                  variant="link"
                >
                  Theme
                </div>
                <div
                  aria-expanded={null}
                  className="collapse show"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/theming/preview-and-colors"
                        onClick={[Function]}
                      >
                        Colors
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/logos"
                        onClick={[Function]}
                      >
                        Logos
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/buttons"
                        onClick={[Function]}
                      >
                        Buttons
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/navigation"
                        onClick={[Function]}
                      >
                        Navigation
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/footer"
                        onClick={[Function]}
                      >
                        Footer
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/hero"
                        onClick={[Function]}
                      >
                        Hero
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="card"
              >
                <div
                  className="card-header"
                  onClick={[Function]}
                  variant="link"
                >
                  Custom Pages
                </div>
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/custom-pages/about"
                        onClick={[Function]}
                      >
                        About
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/contact"
                        onClick={[Function]}
                      >
                        Contact
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/donate"
                        onClick={[Function]}
                      >
                        Donate
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/tos"
                        onClick={[Function]}
                      >
                        Terms of Service
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/honor"
                        onClick={[Function]}
                      >
                        Honor Code
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/privacy"
                        onClick={[Function]}
                      >
                        Privacy Policy
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="card"
              >
                <div
                  className="card-header"
                  onClick={[Function]}
                  variant="link"
                >
                  Instance Settings
                </div>
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/settings/general"
                        onClick={[Function]}
                      >
                        General
                      </a>
                      <a
                        aria-current="page"
                        className="disabled active"
                        href="/"
                        onClick={[Function]}
                        style={Object {}}
                      >
                        Domain
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="card"
              >
                <div
                  className="card-header"
                  onClick={[Function]}
                  variant="link"
                >
                  Courses (Studio)
                </div>
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/courses/manage"
                        onClick={[Function]}
                      >
                        Manage
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="pr-0 col-md-9"
          >
            <span>
               Test! 
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
